scripts/dats_jsonld_annotator/annotator.py

```diff
@@ -0,0 +1,323 @@
+        # We are saving this somewhere other than the original folder
+        out_name = f"{dats_json_f.parent.name}_{out_name}"
+    if out_path.is_dir():
+        dats_jsonld_f = out_path / out_name
+    elif out_path.parent.is_dir():
+        # TODO: check if there is a way to explicitly look for a file that doesn't exist
+        logger.info(f"I will write to {out_path.resolve() = }")
+        dats_jsonld_f = out_path.parent / out_name
+    else:
+        raise Exception(
+            f"{out_path = } for {dats_json_f.resolve} is not a path to a file or directory. "
+            f"I don't know how to store the output"
+        )
+    return dats_jsonld_f
+
+
+def dats_to_jsonld(dats_f, schema_f, context_dir, out_path=None, clobber=False):
+    """
+    Helper function to load the inputs and store the annotated DATS.jsonld file
+    """
+    # TODO: log how many terms were not annotated because they are missing from context
+    tic = time.time()
+    dats_jsonld_f = gen_jsonld_outpath(dats_f, out_path)
+    if dats_jsonld_f.is_file():
+        if clobber:
+            logger.warning(
+                f"{dats_jsonld_f.resolve()} already exists and {clobber = }. "
+                f"I will overwrite the file now!"
+            )
+        else:
+            logger.warning(
+                f"{dats_jsonld_f.resolve()} already exists and {clobber = }. "
+                f"Consider setting the clobber flag to overwrite existing files"
+            )
+            return
+
+    dats_json = json.load(open(dats_f))
+    schema = json.load(open(schema_f))
+
+    # TODO validate the DATS.json before annotation
+    try:
+        dats_jsonld, context = annotate_dats_object(dats_json, schema, {}, context_dir)
+    except Exception as e:
+        logger.exception(f"Annotating {dats_f} did not complete!", e, exc_info=True)
+
+    # Prefill the context with the SDO mapping
+    context["sdo"] = "https://schema.org/"
+    # Combine the context and the dats graph
+    dats_jsonld["@context"] = [
+        context,
+    ]
+    logger.info(
+        f"Final result written to {dats_jsonld_f.resolve()}! This took {time.time()-tic :.2f} seconds"
+    )
+    json.dump(dats_jsonld, open(dats_jsonld_f, "w"), indent=2)
+
+
+def main(cli_args):
+
+    logging.basicConfig(format="%(levelname)s:  %(message)s", level=logging.INFO)
+    parser = argparse.ArgumentParser(
+        description="Annotate a DATS.json file to DATS.jsonld given a schema"
+    )
+    parser.add_argument(
+        "dats_path",
+        type=pal.Path,
+        help="""
+                If this is a path to a DATS file, then only this DATS file is annotated.
+                If this is a path to a directory, then each subdirectory of this directory is expected
+                to contain a DATS file called DATS.json. Each of these files will then be annotated iteratively
+                """,
+    )
+    parser.add_argument(
+        "-ds",
+        "--dats_schema",
+        type=pal.Path,
```

Comment:
should there be a help text here?

Reply:
Yes indeed. Thanks for finding this! I will add a help string.